src/Acme.UI/Controllers/Authentication/MemberController.cs

```diff
@@ -67,5 +78,75 @@ public ActionResult SubmitLogout(string returnUrl)
             FormsAuthentication.SignOut();
             return Redirect(returnUrl);
         }
+
+        public ActionResult RenderRegistration()
+        {
+            var register= this.Umbraco.AssignedContentItem as Register;
+
+            var model = new MemberViewModel {
+                ConfirmPasswordTitle = register.ConfirmPasswordTitle,
+                DisplayNameTitle = register.DisplayNameTitle,
+                EmailTitle = register.EmailTitle,
+                PasswordTitle = register.PasswordTitle,
+                UserNameTitle = register.UserNameTitle,
+                AddressTitle = register.AddressTitle,
+                PhoneTitle = register.PhoneTitle
+            };
+
+            return PartialView("~/Views/BusinessPages/Authentication/_Registration.cshtml", model);
+        }
+
+        public ActionResult SubmitRegistration(MemberViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                // check if user exists already
+                var member = Services.MemberService.FindByUsername(model.UserName, 0, 1, out var totalRecs);
```

Comment:
I think you can get rid of the whole block of code that finds and checks.  (104 - 109)
This will remove the whole if else conditions too
The block below will raise an error if the users exists right?